source/custom/visa_service.custom.cpp

```diff
@@ -34,9 +45,18 @@ class DriverErrorException : public std::runtime_error {
   }
 };
 
+static void ServerObjectCleanup(visa_grpc::VisaService::LibrarySharedPtr library)
+{
+  std::unique_lock<std::shared_mutex> lock(s_server_object_lock);
+  library->Close(s_defaultRM);
+  s_defaultRM = VI_NULL;
+}
```

Comment:
I also need to clear `s_async_operations` from here. I'll do that on the next commit. While these operations wouldn't leak under normal circumstances, it could happen if the client crashes between starting and finishing an operation.